spec/acceptance/iis_application_spec.rb

```diff
@@ -80,6 +80,72 @@
         remove_all_sites
       end
     end
+
+    context 'with nested virtual directory' do
+      before(:all) do
+        @site_name  = SecureRandom.hex(10)
+        @app_name   = SecureRandom.hex(10)
+        create_site(@site_name, true)
+        create_path("c:\\inetpub\\wwwroot\\subFolder\\#{@app_name}")
+        @manifest = <<-HERE
+          iis_application{'subFolder/#{@app_name}':
+            ensure => 'present',
+            applicationname => 'subFolder/#{@app_name}',
+            physicalpath => 'c:\\inetpub\\wwwroot\\subFolder\\#{@app_name}',
+            sitename => '#{@site_name}'
+          }
+        HERE
+      end
+
+      it_behaves_like 'an idempotent resource'
+
+      describe "application validation" do
```

Comment:
If I don't put a ```describe``` block here, then the ```it_behaves_like``` will actually fire after the ```it``` below here. Since the ```it_behaves_like``` is the function that actually runs the manifest, that meant all of the tests in the ```it``` failed. Putting the tests in a ```describe``` was the only way I found to ensure the tests executed in the proper order.